### PR TITLE
Ports Looping Sounds

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Port TG's looping sounds datum, along with subsequent updates: https://github.com/tgstation/tgstation/pull/31773

Looping sounds can be attached to objects and they will, as their name implies, loop until something shuts them off (or the object is destroyed). 

You can normally set a sound to repeat indefinitely, but the problem with this is that it's never going to change direction or volume based on distance. With looping sounds, you can have short loops that repeat rapidly that can emulate the sound getting louder/quieter as you get closer/step away. This also allows for directionality, as well.

You can also customize looping sounds to have a start up and end sound, in addition to the sound that loops.

Currently, this is system is only in place for showers.

Also makes minor tweaks to the default `playsound` by utilizing proper defines and adding support for sound blocking by walls.

:cl: Fox McCloud
tweak: showers are a bit more immersive, with realistic shower-like sounds™
/:cl:

### CODE REVIEW

**Comment by Spartan6** [comment]
The possibilities...

File: code/game/objects/structures/watercloset.dm [hunk 1]
```diff
@@ -266,6 +268,7 @@
 /obj/machinery/shower/Destroy()
 	if(mymist)
```
Comment (Fox-McCloud) [inline code/game/objects/structures/watercloset.dm]: Heresy

**Comment by KasparoVy** [comment]
Now we can add the kamikazes from SS.
https://www.youtube.com/watch?v=ydxIeO-2C2M

**Comment by taukausanake** [comment]
It would be nice to get the fire alarm sound to work with this. I don't think we have a sound as it is anymore?

**Comment by Fox-McCloud** [comment]
We never have.

Goon did, but not us.

**Pull request merged**